### PR TITLE
Remove ununsed prealloc frames

### LINKED ISSUE #2298: Declared but unused variable `aPreAllocFrames`
@reuben -- the `aPreAllocFrames` variable in `deepspeech.cc` is declared but unused, unless I'm missing something:

https://github.com/mozilla/DeepSpeech/blob/29836c9dcc4c2e30e57ae6e50d303fbb317325bb/native_client/deepspeech.cc#L332

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #2298

### CODE REVIEW

File: native_client/dotnet/DeepSpeechClient/DeepSpeech.cs [hunk 1]
```diff
@@ -204,14 +204,11 @@ public unsafe void PrintVersions()
         /// <summary>
         /// Creates a new streaming inference state.
         /// </summary>
-        /// <param name="aPreAllocFrames">Number of timestep frames to reserve.
-        /// One timestep is equivalent to two window lengths(20ms).
-        /// If set to 0 we reserve enough frames for 3 seconds of audio(150).</param>
         /// <param name="aSampleRate">The sample-rate of the audio signal</param>
         /// <exception cref="ArgumentException">Thrown when the native binary failed to initialize the streaming mode.</exception>
-        public unsafe void SetupStream(uint aPreAllocFrames, uint aSampleRate)
```
Comment (carlfm01) [inline native_client/dotnet/DeepSpeechClient/DeepSpeech.cs]: @lissyx 

https://github.com/mozilla/DeepSpeech/blob/b92cea1c0b38bb94f25585a4c8f970287a37caeb/examples/net_framework/DeepSpeechWPF/MainWindow.xaml.cs#L255 now only needs the sample rate, will fail with the extra 0

Comment (lissyx) [reply native_client/dotnet/DeepSpeechClient/DeepSpeech.cs]: We don't build that? I thought we did, I checked twice for SetupStream calls 😕

Comment (lissyx) [reply native_client/dotnet/DeepSpeechClient/DeepSpeech.cs]: Good catch anyway, thanks

Comment (carlfm01) [reply native_client/dotnet/DeepSpeechClient/DeepSpeech.cs]: @lissyx 

> 
> 
> We don't build that? I thought we did, I checked twice for SetupStream calls 😕
We don't build the WPF example : 
https://github.com/mozilla/DeepSpeech/blob/69ad6016a7e991138959f079c4067769820abacf/taskcluster/tc-tests-utils.sh#L879-L916

I will try to add a window on my time and add the example to the builds, should be straightforward.

**Comment by reuben** [comment]
Do we really want to break backwards compat in the C API/ABI with this change? Clients compiled against older versions will run but crash mysteriously. I had a PR doing a bunch of cleaning up on the API, but it got held back by having to figure out what compatibility strategy we want to have. Technically since we're 0.x anything goes, but we've tried to keep things stable when possible. Thoughts?

**Comment by lissyx** [comment]
> Do we really want to break backwards compat in the C API/ABI with this change? Clients compiled against older versions will run but crash mysteriously. I had a PR doing a bunch of cleaning up on the API, but it got held back by having to figure out what compatibility strategy we want to have. Technically since we're 0.x anything goes, but we've tried to keep things stable when possible. Thoughts?

I do agree, but I'm also thinking that it's not a good idea we keep useless parameter for nothing. For now, I think the strategy we should have is do not carry any burden we absolutely don't have to: we'll always end up having to keep crap, so we should try and limit as much as possible :)

**Pull request merged**